### ISSUE TITLE
Unable to use custom FieldProcessors in List/Map

### ISSUE THREAD

**Issue by VriesDeRick** [Jaguar-dart/jaguar_serializer#73]
For example, in our codebase we have the following UriProcessor:

```dart
class UriProcessor implements FieldProcessor<dynamic, String> {
  const UriProcessor();

  @override
  dynamic deserialize(String value) {
    if (value == null) {
      return null;
    }

    try {
      return Uri.parse(value);
    } on FormatException {
      return null;
    }
  }

  @override
  String serialize(dynamic value) {
    return value.toString();
  }
}

```

It's just a simple serializer. However, we cannot use it for Object properties like `List<Uri>` or `Map<String, Uri>`, since the Processor can only encode/decode a simple Uri.

**Comment by lejard-h** [comment]
Not possible for now,
as a workaround, you can create a processor for a `List<Uri>`

```
class ListUriProcessor implements FieldProcessor<List<Uri>, List<String>> {
  const uriProcessor = const UriProcessor();
  const ListUriProcessor();

  @override
  dynamic deserialize(List<String> value) {
    if (value == null || value.isEmpty) {
      return value;
    }

    return value.map((m) => uriProcessor.deserialize(m)).toList();
  }

  @override
  List<String> serialize(List<Uri> value) {
    return value?.map((m) => uriProcessor.serialize(m))?.toList();
  }
}
```

**Comment by lejard-h** [comment]
but did you try to apply the processor on the list field,

```dart
@GenSerializer(
  fields: const {
    "myList": const Property(processor: const UriProcessor())
  }
```

**Comment by VriesDeRick** [comment]
I did, but the generated code just results in the following (`thumbnails` is the field which is a `List<Uri>`)
```dart 
model.thumbnails = thumbnailsUriProcessor.deserialize(map["thumbnails"]);
```
The problem is that it immediately calls the processor, instead of first checking `map["thumbnails]` for being a List or a Map

**Comment by lejard-h** [comment]
fix in next release `1.0.0-dev2`
